runtime/doc/lua.txt

```diff
@@ -2230,6 +2230,12 @@ vim.tbl_count({t})                                           *vim.tbl_count()*
 vim.tbl_deep_extend({behavior}, {...})                 *vim.tbl_deep_extend()*
     Merges recursively two or more tables.
 
+    Only values that are empty tables or tables that are not |lua-list|s
+    (indexed by consecutive integers starting from 1) are merged recursively.
```

Comment:
if we are talking about "merging recursively", maybe it's not worth mentioning empty tables?

```suggestion
    Only tables that are not |lua-list|s
    (indexed by consecutive integers starting from 1) are merged recursively.
```

Reply:
No, these are special cased (_and_ list-like), so this is important to call out.

Reply:
The current "values that are ..." is a way to document [this behavior](https://github.com/neovim/neovim/pull/30309#discussion_r1749287011).